pyquil/paulis.py

```diff
@@ -126,7 +132,10 @@ class PauliTerm(object):
     """
 
     def __init__(
-        self, op: str, index: PauliTargetDesignator, coefficient: ExpressionDesignator = 1.0
+        self,
+        op: str,
+        index: Optional[PauliTargetDesignator],
```

Comment:
Why is there an extra `Optional[]` here?

Reply:
because for `sI()` the qubit argument is optional, so this can actually be `None`

Reply:
Shouldn't it be `index: Optional[PauliTargetDesignator] = None` then?

Reply:
Not necessarily. Despite the name, it's valid (but rare) to have an `Optional` required argument. Adding the `= None` makes it a keyword arg, which means it gets a default value if the caller doesn't explicitly provide it. Whereas `Optional` without `= None` means you must explicitly pass `None` when you want `index` to be `None`.

Making `index` a keyword arg here should be backwards compatible, but since it was a required arg before adding the type hints, it seems reasonable for it to remain one after.